sql/src/test/java/io/crate/analyze/SelectStatementAnalyzerTest.java

```diff
@@ -384,6 +389,44 @@ public void testAmbiguousOrderByOnAlias() throws Exception {
     }
 
     @Test
+    public void selectGroupByOrderByWithColumnMissingFromSelect() throws Exception {
+        expectedException.expect(UnsupportedOperationException.class);
+        expectedException.expectMessage("ORDER BY field 'id' must appear in the select clause " +
+                                        "when grouping or global aggregation is used.");
+        analyze("select name, count(id) from users group by name order by id");
+    }
+
+    @Test
+    public void selectGlobalAggregationOrderByWithColumnMissingFromSelect() throws Exception {
+        expectedException.expect(UnsupportedOperationException.class);
+        expectedException.expectMessage("ORDER BY field 'id' must appear in the select clause " +
+                                        "when grouping or global aggregation is used.");
+        analyze("select count(id) from users order by id");
+    }
+
+    @Test
+    public void selectGroupByOrderByWithAggragateFunctionInOrderByClause() throws Exception {
+        expectedException.expect(UnsupportedOperationException.class);
+        expectedException.expectMessage("ORDER BY function 'max(count(upper(name)))' is not allowed. Only scalar functions can be used.");
+        analyze("select name, count(id) from users group by name order by max(count(upper(name)))");
+    }
+
+    @Test
+    public void selectGroupByOrderBy() throws Exception {
```

Comment:
please indicate somehow that this test is testing the validation, since there is no assertion within the body it is unclear what this test is doing (at least without the context of this PR)